examples/ensemble/plot_adaboost_multiclass.py

```diff
@@ -68,18 +82,36 @@
     real_test_errors.append(1.0 - accuracy_score(real_test_predict, y_test))
     discrete_test_errors.append(1.0 - accuracy_score(discrete_test_predict, y_test))
 
+# %%
```

Comment:
For the rest of the example, I would restrict to the SAMME algorithm as in the paper:

```python
# %%
weak_learners_info = pd.DataFrame({
    "Number of trees": range(1, n_estimators + 1),
    "Errors": bdt_discrete.estimator_errors_,
    "Weights": bdt_discrete.estimator_weights_,
}).set_index("Number of trees")

axs = weak_learners_info.plot(
    subplots=True, layout=(1, 2), figsize=(10, 4), legend=False, color="tab:blue"
)
axs[0, 0].set_ylabel("Train error")
axs[0, 0].set_title("Weak learners training error")
axs[0, 1].set_ylabel("Weight")
axs[0, 1].set_title("Weak learners weight")
fig = axs[0, 0].get_figure()
fig.suptitle("Weak learners errors and weights for SAMME")
fig.tight_layout()
```

This should show the following:

![image](https://github.com/scikit-learn/scikit-learn/assets/7454015/c1669b4a-762f-4884-9ede-8b4fdd591c35)

As in the paper, we can discuss that the weak learner's error and weight have the opposite behaviour. The error is increasing overtime since the algorithm will try to better classify difficult examples and therefore will make mistakes on the majority of the samples. Thus to compensate for this effect, the weights affected will be lower for the latest weak learners that are more error-prone in general.